src/pages/index.js

```diff
@@ -0,0 +1,296 @@
+
+// Preview image popup elements
+const previewModal = document.querySelector("#preview-modal");
+const previewModalImageEl = previewModal.querySelector(".modal__image");
+const previewModalCaptionEl = previewModal.querySelector(".modal__caption");
+const previewModalCloseEl = previewModal.querySelector(".modal__close-btn");
+
+// Card related elements
+const cardTemplate = document.querySelector("#card-template");
+const cardslist = document.querySelector(".cards__list");
+
+let selectedCard, selectedCardId;
+
+function getCardElement(data) {
+  const cardElement = cardTemplate.content
+    .querySelector(".card")
+    .cloneNode(true);
+
+  // Select elements within the card
+  const cardNameEl = cardElement.querySelector(".card__title");
+  const cardImageEl = cardElement.querySelector(".card__image");
+  const cardLikedBtn = cardElement.querySelector(".card__like-button");
+  const cardDeleteBtn = cardElement.querySelector(".card__delete-btn");
+
+  // set card details
+  cardNameEl.textContent = data.name;
+  console.log(data);
+  cardImageEl.src = data.link;
+  cardImageEl.alt = data.name;
+
+  // Handle like button click
+  cardLikedBtn.addEventListener("click", (evt) =>
+    handleLike(evt, data._id, data.isLiked)
+  );
+  if (data.isLiked) {
+    cardLikedBtn.classList.add("card__like-button_liked");
+  }
+
+  // Handle delete button click
+  cardDeleteBtn.addEventListener("click", () =>
+    handleDeleteCard(cardElement, data._id)
+  );
+
+  // Handle image preview modals
+  cardImageEl.addEventListener("click", () => {
+    previewModalImageEl.src = data.link;
+    previewModalImageEl.alt = data.name;
+    previewModalCaptionEl.textContent = data.name;
+    openModal(previewModal);
+  });
+
+  return cardElement;
+}
+
+function handleClickOverlay(evt) {
+  if (evt.target.classList.contains("modal_opened")) {
+    closeModal(evt.target);
+  }
+}
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keyup", handleEscClose);
+  document.addEventListener("click", handleClickOverlay);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keyup", handleEscClose);
+  document.removeEventListener("click", handleClickOverlay);
+}
+
+function handleEscClose(evt) {
+  if (evt.key === "Escape") {
+    const activeModal = document.querySelector(".modal_opened");
+    closeModal(activeModal);
+  }
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+  const editProfilesubmitButton = evt.submitter;
+  setButtonText(editProfileSubmitButton, true);
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDesriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setTimeout(() => {
```

Comment:
`setTimeout` is not needed in the project because there should be **natural** response waiting